### PR TITLE
Added support for traceparent header

### PR DESCRIPTION
**Pull request by robert-smartbear**

### CODE REVIEW

File: lib/bugsnag_http_client.dart [hunk 1]
```diff
@@ -110,10 +125,21 @@ class Client extends http.BaseClient{
   }
 
   @override
-  Future<http.Response> post(Uri url, {Map<String, String>? headers, Object? body, Encoding? encoding})  async {
-    var requestId = _sendRequestStartNotification(url.toString(),"POST");
+  Future<http.Response> post(Uri url,
+      {Map<String, String>? headers, Object? body, Encoding? encoding}) async {
+    var requestId = _sendRequestStartNotification(url.toString(), "POST");
```
Comment (lemnik) [inline lib/bugsnag_http_client.dart]: nit: these all look like they can be `final` rather than `var`.

File: lib/bugsnag_http_client.dart [hunk 2]
```diff
@@ -208,19 +294,22 @@ class Client extends http.BaseClient{
...
+            requestId,
+            http.Response.bytes(
+              bytes,
+              streamedResponse.statusCode,
+              request: request,
+              headers: streamedResponse.headers,
+              isRedirect: streamedResponse.isRedirect,
+              persistentConnection: streamedResponse.persistentConnection,
+              reasonPhrase: streamedResponse.reasonPhrase,
+            ));
```
Comment (lemnik) [inline lib/bugsnag_http_client.dart]: nit: formatting

```suggestion
        _sendRequestCompleteNotification(
            requestId,
            http.Response.bytes(
              bytes,
              streamedResponse.statusCode,
              request: request,
              headers: streamedResponse.headers,
              isRedirect: streamedResponse.isRedirect,
              persistentConnection: streamedResponse.persistentConnection,
              reasonPhrase: streamedResponse.reasonPhrase,
            ),
        );
```

File: lib/bugsnag_http_client.dart [hunk 3]
```diff
@@ -189,9 +266,18 @@ class Client extends http.BaseClient{
...
-      var response = await get(url, headers: headers);
+      var allHeaders = headers ?? {};
+      _headersProvider
+          .requestHeaders(
+        url: url.toString(),
+        requestId: requestId,
+      )
+          ?.forEach((key, value) {
+        allHeaders[key] = value;
+      });
```
Comment (lemnik) [inline lib/bugsnag_http_client.dart]: Could we not use the spread operator here:

```suggestion
     final allHeaders = {
          ...?headers,
          ...?(_headersProvider.requestHeaders(
                  url: url.toString(),
                  requestId: requestId,
              ),
          ),
      };
```

**Review by richardelms — APPROVED** [review]
👍

**Review by lemnik — APPROVED** [review]

**Pull request merged**